modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2703,9 +2705,25 @@ private CompletableFuture<Void> destroyPartitionStorages(TablePartitionId tableP
             destroyFutures.add(runAsync(() -> internalTable.txStateStorage().destroyTxStateStorage(partitionId), ioExecutor));
         }
 
+        destroyFutures.add(
+                supplyAsync(() -> destroyReplicationProtocolStorages(tablePartitionId, table), ioExecutor).thenCompose(identity())
+        );
+
         return allOf(destroyFutures.toArray(new CompletableFuture[]{}));
     }
 
+    private CompletableFuture<Object> destroyReplicationProtocolStorages(TablePartitionId tablePartitionId, TableImpl table) {
```

Comment:
Why does this method return a `CompletableFuture`?

Reply:
I used the future to communicate an exception to the caller. This looks like an abuse, so now I changed it to wrapping the checked exception in `IgniteInternalException`